### PR TITLE
Fixing of bugs in Baghchal

### PR DESCRIPTION
**Pull request by Suraj1127**
* Fixed bug of alpha beta pruning and minimax algorithm.
* Worked on uiconfig and its integration into the program.

### CODE REVIEW

File: ui.py [hunk 1]
```diff
@@ -215,50 +214,54 @@ def check_win(self):
...
-
-        # if self.config.has_option('game', 'aistrength'):
-        #     cmdline.append(self.config.get('game', 'aistrength'))
-        # else:
-        #     cmdline.append('3')
+        self.config = configparser.ConfigParser()
+        self.config.read(os.path.expanduser('uiconf'))
+
+        if self.config.has_option('game', 'ai'):
+            if self.config.get('game', 'ai').lower() == 'sheep':
```
Comment (code-geek) [inline ui.py]: are we using `sheep` or `goats` in the rest of the code?

File: ui.py [hunk 2]
```diff
@@ -215,50 +214,54 @@ def check_win(self):
...
 
     def make_ai_move(self, ev=None):
         move = self.engine.get_best_move()
         self.apply_move(move)
 
 
 def configure():
-    config = configparser.SafeConfigParser()
+    config = configparser.ConfigParser()
+    print('haha')
```
Comment (code-geek) [inline ui.py]: Let's remove the print statement. :smile: 
print('Lol')

File: ui.py [hunk 3]
```diff
@@ -215,50 +214,54 @@ def check_win(self):
             # self.game = None
             self.statustext.set('Tigers win!')
             self.win = 'tigers'
+            print(self.win)
```
Comment (code-geek) [inline ui.py]: Let's remove the print statements everywhere.

**Comment by Suraj1127** [comment]
The bugs of AI engine have been solved.

**Pull request merged**